tests/rkt_tests.go

```diff
@@ -515,3 +555,189 @@ func checkAppStatus(t *testing.T, ctx *rktRunCtx, multiApps bool, appName, expec
+	manifest []byte
+}
+
+// parsePodInfo parses the 'rkt status $UUID' result into podInfo struct.
+// For example, the 'result' can be:
+// state=running
+// networks=default:ip4=172.16.28.103
+// pid=14352
+// exited=false
+func parsePodInfoOutput(t *testing.T, result string, p *podInfo) {
+	lines := strings.Split(strings.TrimSpace(result), "\n")
+	for _, line := range lines {
+		tuples := strings.SplitN(line, "=", 2)
+		if len(tuples) != 2 {
+			t.Fatalf("Unexpected line: %v", line)
+		}
+
+		switch tuples[0] {
+		case "state":
+			p.state = tuples[1]
+		case "networks":
+			networks := strings.Split(tuples[1], ",")
+			for _, n := range networks {
+				fields := strings.Split(n, ":")
+				if len(fields) != 2 {
+					t.Fatalf("Unexpected network info format: %v", n)
+				}
+
+				ip4 := strings.Split(fields[1], "=")
+				if len(ip4) != 2 {
+					t.Fatalf("Unexpected network info format: %v", n)
+				}
+
+				p.networks = append(p.networks, &networkInfo{
+					name: fields[0],
+					ipv4: ip4[1],
+				})
+			}
+		case "pid":
+			pid, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				t.Fatalf("Cannot parse the pod's pid %q: %v", tuples[1], err)
+			}
+			p.pid = pid
+		}
+		if strings.HasPrefix(tuples[0], "app-") {
+			exitCode, err := strconv.Atoi(tuples[1])
+			if err != nil {
+				t.Fatalf("cannot parse exit code from %q : %v", tuples[1], err)
+			}
+			appName := strings.TrimPrefix(tuples[0], "app-")
+
+			for _, app := range p.apps {
+				if app.name == appName {
+					app.exitCode = exitCode
+					break
+				}
+			}
+		}
+	}
+}
+
+func getPodDir(t *testing.T, ctx *rktRunCtx, podID string) string {
+	podsDir := path.Join(ctx.dataDir(), "pods")
+
+	dirs, err := ioutil.ReadDir(podsDir)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	for _, dir := range dirs {
+		podDir := path.Join(podsDir, dir.Name(), podID)
+		if _, err := os.Stat(podDir); err == nil {
+			return podDir
+		}
+	}
+	t.Fatalf("Failed to find pod directory for pod %q", podID)
+	return ""
+}
+
+// getPodInfo returns the pod info for the given pod ID.
+func getPodInfo(t *testing.T, ctx *rktRunCtx, podID string) *podInfo {
+	p := &podInfo{id: podID}
+
+	// Read pod manifest.
+	podManifestPath := path.Join(getPodDir(t, ctx, podID), "pod")
+	f, err := os.Open(podManifestPath)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	p.manifest, err = ioutil.ReadAll(f)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Fill app infos.
+	var manifest schema.PodManifest
+	if err := json.Unmarshal(p.manifest, &manifest); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	for _, app := range manifest.Apps {
+		p.apps = append(p.apps, &appInfo{
+			name: app.Name.String(),
+			// TODO(yifan): Get the image's name.
+			image: &imageInfo{id: app.Image.ID.String()},
+		})
+	}
+
+	// Fill other infos.
+	output, err := exec.Command("/bin/bash", "-c", fmt.Sprintf("%s status %s", ctx.cmd(), podID)).CombinedOutput()
```

Comment:
@krnowak Then I need to split the `ctx.cmd()` into `[]string`. Also not using `gexpect` here because I want to get the whole output.